src/tool_help.c

```diff
@@ -863,16 +864,64 @@ static const struct feat feats[] = {
   {"alt-svc",        CURL_VERSION_ALTSVC},
 };
 
-void tool_help(void)
+static void print_category(curlhelp_t category)
+{
+  int i;
+  for(i = 0; helptext[i].opt; ++i)
+    if(helptext[i].categories & category) {
+      printf(" %-19s %s\n", helptext[i].opt, helptext[i].desc);
+    }
+}
+
+/* Prints category if found. If not, it returns 1 */
+static int get_category_content(const char *category)
+{
+  int i;
+  for(i = 0; categories[i].opt; ++i)
+    if(strcmp(categories[i].opt, category) == 0) {
+      printf("%s: %s\n", categories[i].opt, categories[i].desc);
+      print_category(categories[i].category);
+      return 0;
+    }
+  return 1;
+}
+
+/* Prints all categories and their description */
+static void get_categories()
 {
   int i;
+  for(i = 0; categories[i].opt; ++i)
+    printf(" %-19s %s\n", categories[i].opt, categories[i].desc);
+}
+
+void tool_help(const char *category)
+{
   puts("Usage: curl [options...] <url>");
-  for(i = 0; helptext[i].opt; i++) {
-    printf(" %-19s %s\n", helptext[i].opt, helptext[i].desc);
-#ifdef PRINT_LINES_PAUSE
-    if(i && ((i % PRINT_LINES_PAUSE) == 0))
-      tool_pressanykey();
-#endif
+  /* If no category was provided */
+  if(!category) {
+    const char *category_note = "\nThis is not the full help, this "
+    "menu is stripped into categories.\nUse \"--help category\" to get "
+    "an overview of all categories.\nFor all options use the manual"
+    " or \"--help all\".";
+    print_category(CURLHELP_IMPORTANT);
+    puts(category_note);
+    return;
+  }
+  /* Lets print everything if "all" was provided */
+  if(strcmp(category, "all") == 0) {
```

Comment:
We usually write these comparisons as `if(!strcmp())` ... (ie we avoid comparisons with 0)

Also, should we do the name check case insensitive perhaps?